core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -461,7 +462,7 @@ public synchronized boolean isReady() {
 
       @Override
       public void cancel(Status reason) {
-        if (!internalCancel(reason)) {
+        if (!internalCancel(stripCause(reason))) {
```

Comment:
This is `InProcessClientStream.cancel(...)`, which invokes `serverStreamListener.closed`. Therefore stripped.